src/cargo/core/shell.rs

```diff
@@ -143,6 +143,27 @@ impl Shell {
         }
     }
 
+    /// Print the message to standard output, where the query will have the color `color`.
+    pub fn print_stdout_with_part_color(
+        &mut self,
+        name_no_query: Vec<&str>,
+        desc_no_query: Vec<&str>,
+        query: &str,
+        color: Color,
+        justified: bool,
+    ) -> CargoResult<()> {
+        match self.verbosity {
```

Comment:
Wouldn't this mean that `cargo search -q foo` wouldn't actually print anything?

Reply:
thank you,done.